sdks/java/io/jms/src/test/java/org/apache/beam/sdk/io/jms/JmsIOTest.java

```diff
@@ -124,46 +127,56 @@ public class JmsIOTest {
       RetryConfiguration.create(1, Duration.standardSeconds(1), null);
   @Rule public final transient TestPipeline pipeline = TestPipeline.create();
 
-  @Parameterized.Parameters(name = "with client class {3}")
+  private static Stream<List<Object>> connectionFactoriesParams() {
+    return Stream.of(
+        Arrays.asList(
+            "vm://localhost", 5672, "jms.sendAcksAsync=false", ActiveMQConnectionFactory.class),
+        Arrays.asList(
+            "amqp://localhost", 5672, "jms.forceAsyncAcks=false", JmsConnectionFactory.class));
+  }
+
+  @Parameterized.Parameters(name = "with client class {3}, use provider fn? {4}")
```

Comment:
will this double the number of tests?

We won't simply 2x number of test, which might become infeasible as number of test parameter increases, but will just add one more test case with this parameter changed.

I think we can either add one test case to ActiveMQConnectionFactory, or add one or a few tests testing connectionFactoryFn, while keep most existing tests untouched